examples/triangle_xform/readme.md

```diff
@@ -0,0 +1,58 @@
+# Transforming triangle vertices using a transformation matrix
+
+## Description
+
+This example contains the SIMD implementation of a common computer graphics task:
+transforming vertices with a matrix.
+
+## Implementation
+
+There are two implementations:
+
+- scalar version, uses an array-of-structures layout, where each triangle contains
+  three vertices, and each vertex contains only a 3D position vector; the algorithm
+  operates on **one triangle at a time**.
+
+- SIMD version, uses a structure-of-arrays layout, where the structure contains, for
+  each of the X, Y, and Z components of a 3D vector, an array of their values; the
+  algorithm operates on **up to N triangles at once**, where N is number of lanes in a
+  SIMD register.
```

Comment:
I'll try to review all of this tomorrow.

---

This is a bit off-topic but last year I had to handle very large triangle meshes in one project (billions of triangles) and what I ended up doing was storing three integers per triangle, that index into a SoA of points (x,y,z in SoA form), and doing the bulk transformations in the SoA of points all at once. This loaded every vertex from memory only once even if it was shared by many triangles, and operated on 8 vertices at once using SIMD even though the vertices were often from unrelated triangles.

Reply:
Whoa, that's awesome! My usage of SIMD is way more mundane :)
I hope we can get some of these ideas into the perf book. The reason I submited this example is to have something to give as an example in the area on vertical-vs-horizontal operations and data layouts.

Reply:
So one big difference here is that AFAICT the triangles here are disjoint, that is, no vertex is shared across triangles, so something like this would not improve much perf in this case. 

In my case, it cut the memory bandwidth by ~3, so it pretty much delivered an almost 3x speed-up.